bash_it.sh

```diff
@@ -1,117 +1,129 @@
 #!/usr/bin/env bash
-# Initialize Bash It
-BASH_IT_LOG_PREFIX="core: main: "
 
 # Only set $BASH_IT if it's not already set
-if [ -z "$BASH_IT" ]; then
+if [[ -z "$BASH_IT" ]]; then
 	# Setting $BASH to maintain backwards compatibility
 	export BASH_IT=$BASH
 	BASH="$(bash -c 'echo $BASH')"
 	export BASH
 	BASH_IT_OLD_BASH_SETUP=true
 fi
 
-# Load composure first, so we support function metadata
+# Load composure first, to support function metadata, then create our custom attributes.
 # shellcheck disable=SC1090
 source "${BASH_IT}"/vendor/github.com/erichs/composure/composure.sh
+cite _about _param _example _group _author _version
+cite about-alias about-completion about-plugin
 
-# We need to load logging module first as well in order to be able to log
+# Next, load our logging library so we can give useful feedback to the user
 # shellcheck source=./lib/log.bash
 source "${BASH_IT}/lib/log.bash"
 
-# We can only log it now
-[ -z "$BASH_IT_OLD_BASH_SETUP" ] || _log_warning "BASH_IT variable not initialized, please upgrade your bash-it version and reinstall it!"
+BASH_IT_LOG_PREFIX='core: main: '
+
+# Check for old installations
+if [[ -n "$BASH_IT_OLD_BASH_SETUP" ]]; then
+	_log_warning "BASH_IT variable not initialized, please upgrade your bash-it version and reinstall it!"
+fi
 
 # For backwards compatibility, look in old BASH_THEME location
-if [ -z "$BASH_IT_THEME" ]; then
+if [[ -z "$BASH_IT_THEME" ]]; then
 	_log_warning "BASH_IT_THEME variable not initialized, please upgrade your bash-it version and reinstall it!"
 	export BASH_IT_THEME="$BASH_THEME"
 	unset BASH_THEME
 fi
 
-# support 'plumbing' metadata
-cite _about _param _example _group _author _version
-cite about-alias about-plugin about-completion
-
-# libraries, but skip appearance (themes) for now
-_log_debug "Loading libraries(except appearance)..."
-LIB="${BASH_IT}/lib/*.bash"
-APPEARANCE_LIB="${BASH_IT}/lib/appearance.bash"
-for _bash_it_config_file in $LIB; do
-	if [ "$_bash_it_config_file" != "$APPEARANCE_LIB" ]; then
-		filename=${_bash_it_config_file##*/}
-		filename=${filename%.bash}
-		BASH_IT_LOG_PREFIX="lib: ${filename}: "
-		_log_debug "Loading library file..."
-		# shellcheck disable=SC1090
-		source "$_bash_it_config_file"
-	fi
-done
+# Load the base libraries ...
+# Note: Ordering is important and intentional
 
-# Load vendors
-BASH_IT_LOG_PREFIX="vendor: "
-for _bash_it_vendor_init in "${BASH_IT}"/vendor/init.d/*.bash; do
-	_log_debug "Loading \"$(basename "${_bash_it_vendor_init}" .bash)\"..."
-	# shellcheck disable=SC1090
-	source "${_bash_it_vendor_init}"
-done
-unset _bash_it_vendor_init
+# "utilities" - internal functions meant for use within the bash-it codebase
+BASH_IT_LOG_PREFIX='lib: utilities: '
+_log_debug 'Loading library file ...'
```

Comment:
I think you should create an helper function which changes the prefix, logs, and sources the file, as this pattern is really duplicated here

Reply:
Where would you suggest that live? I'm trying to avoid adding functions to this specific file because load order issues and confusion, as highlighted above.

Reply:
hmm, I am not sure this belongs to a followup PR, this is a lot of code dup that can be prevented, I am okay with defining it in this file or in `utilities.bash` and just load it first. This is a bugfix, but it changes the way we load quite a lot, and I prefer we do it as carefully as we can

Reply:
I strongly discourage ever `source`ing from within a function. Functions introduce scope, which means that anything `declare`d or `typeset` is lost *inside* the function. 

But a helper which changes prefix and logs sounds like it would do well inside `lib/log`.